### PR TITLE
Allow duplicate oneagent arguments

### PR DESCRIPTION
**Pull request by StefanHauth**
## Description

With a previous overhaul of how OneAgent arguments are handled it was made impossible to use an argument multiple times. For some arguments that's reasonable, for others not so much.

This PR enables using the same argument multiple times for the OneAgent, while keeping some more sensible arguments to being used only once. Only those where the operator potentially uses default values are considered. Namely those which can't be used multiple times are:
* `--set-server`
* `--set-tenant`
* `--set-host-id-source`
* `--set-host-group`
* `--set-network-zone`
* `--set-proxy`

## How can this be tested?

Deploy the operator.
Deploy a DynaKube where you set custom one agent arguments. Use especially multiple `--set-host-property` and `--set-host-tag` entries. They should be usable multiple times.

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by chrismuellner — CHANGES_REQUESTED** [review]
Please change the logic as discussed so all arguments are allowed.

We should add a warning when an argument is used that is already configured by the Operator, not block the usage.

**Comment by StefanHauth** [comment]
> Please change the logic as discussed so all arguments are allowed.
> 
> We should add a warning when an argument is used that is already configured by the Operator, not block the usage.

All arguments are allowed, It's just made sure that those, where the operator sets default values, are not duplicated if they are also provided in the Dynakube. In such cases, the operator's default values are overruled by the Dynakube (which always shall take precedence).

**Comment by chrismuellner** [comment]
@StefanHauth I don't see the point of doing that. Let's just leave that logic to the Oneagent.

We allow all values. User provided args are added after operator ones and therefore take priority. If a user provided value is also set by the Operator we only provide a warning and don't remove any existing arg from the list.

**Review by aorcholski — APPROVED** [review]
LGTM

File: pkg/controllers/dynakube/oneagent/daemonset/arguments.go [hunk 1]
```diff
@@ -13,7 +13,11 @@ const customArgumentPriority = 2
 const defaultArgumentPriority = 1
 
 func (dsInfo *builderInfo) arguments() ([]string, error) {
-	argMap := prioritymap.New(prioritymap.WithSeparator(prioritymap.DefaultSeparator), prioritymap.WithPriority(defaultArgumentPriority))
+	argMap := prioritymap.New(
```
Comment (luhi-DT) [inline pkg/controllers/dynakube/oneagent/daemonset/arguments.go:16]: Why do we need a priorityMap at all? Wouldn't it be simpler to just append the args as they are? And not store them in proprietary struct

**Review by luhi-DT — CHANGES_REQUESTED** [review]

File: pkg/controllers/dynakube/oneagent/daemonset/arguments.go (hunk 1 above)
Comment (StefanHauth) [reply pkg/controllers/dynakube/oneagent/daemonset/arguments.go:16]: Maybe, but not in this PR.

Comment (StefanHauth) [reply pkg/controllers/dynakube/oneagent/daemonset/arguments.go:16]: The priority map ensures proper ordering of the arguments, no matter in which order they are added. IMHO this also provides more resilience against inconsiderate changes in the future.

Comment (luhi-DT) [reply pkg/controllers/dynakube/oneagent/daemonset/arguments.go:16]: I am fine with not having it in this bugfix PR, but we should definitely get rid of this logic, its to complicated for what it brings us in terms of benefits

**Review by luhi-DT — APPROVED** [review]

**Review by chrismuellner — APPROVED** [review]
PR description is outdated, but changes were done as expected

**Pull request merged**